torch/distributed/__init__.py

```diff
@@ -82,5 +82,8 @@ def is_available() -> bool:
     #   python test/test_public_bindings.py -k test_correct_module_names
     # working even when USE_DISTRIBUTED=0.  Feel free to add more
     # stubs as necessary.
-    class ProcessGroup:  # type: ignore[no-redef]
+    # We cannot define stubs directly because they confuse pyre
+
+    class _ProcessGroupStub:
         pass
+    sys.modules["torch.distributed"].ProcessGroup = _ProcessGroupStub  # type: ignore[attr-defined]
```

Comment:
by setting sys modules we are able to work around pyre issues? is this only happen when there's no distributed package?

Reply:
Yes, setting `sys.modules` avoid defining another `ProcessGroup` class in this file, which confuses pyre.